extensions/SQLServer/SQLServer/SqlServerMemory.cs

```diff
@@ -0,0 +1,645 @@
+    {
+        index = NormalizeIndexName(index);
+
+        if (!(await this.DoesIndexExistsAsync(index, cancellationToken).ConfigureAwait(false)))
+        {
+            // Index does not exist
+            yield break;
+        }
+
+        string queryColumns = "[key], [payload], [tags]";
+
+        if (withEmbeddings)
+        {
+            queryColumns += ", [embedding]";
+        }
+
+        if (limit < 0)
+        {
+            limit = int.MaxValue;
+        }
+
+        using var connection = new SqlConnection(this._config.ConnectionString);
+        await connection.OpenAsync(cancellationToken).ConfigureAwait(false);
+
+        using (SqlCommand command = connection.CreateCommand())
+        {
+            var tagFilters = new TagCollection();
+
+            command.CommandText = $@"
+            WITH [filters] AS
+		    (
+			    SELECT
+				    cast([filters].[key] AS NVARCHAR(256)) COLLATE SQL_Latin1_General_CP1_CI_AS AS [name],
+				    cast([filters].[value] AS NVARCHAR(256)) COLLATE SQL_Latin1_General_CP1_CI_AS AS [value]
+			    FROM openjson(@filters) [filters]
+		    )
+            SELECT TOP (@limit)
+                {queryColumns}
+            FROM
+                {this.GetFullTableName(this._config.MemoryTableName)}
+		    WHERE 1=1
+            AND {this.GetFullTableName(this._config.MemoryTableName)}.[collection] = @index
+            {this.GenerateFilters(index, command.Parameters, filters)};";
+
+            command.Parameters.AddWithValue("@index", index);
+            command.Parameters.AddWithValue("@limit", limit);
+            command.Parameters.AddWithValue("@filters", JsonSerializer.Serialize(tagFilters));
+
+            using var dataReader = await command.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
+
+            while (await dataReader.ReadAsync(cancellationToken).ConfigureAwait(false))
+            {
+                yield return await this.ReadEntryAsync(dataReader, withEmbeddings, cancellationToken).ConfigureAwait(false);
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetSimilarListAsync(string index, string text, ICollection<MemoryFilter>? filters = null, double minRelevance = 0, int limit = 1, bool withEmbeddings = false, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        index = NormalizeIndexName(index);
+
+        if (!(await this.DoesIndexExistsAsync(index, cancellationToken).ConfigureAwait(false)))
+        {
+            // Index does not exist
+            yield break;
+        }
+
+        Embedding embedding = await this._embeddingGenerator.GenerateEmbeddingAsync(text, cancellationToken).ConfigureAwait(false);
+
+        string queryColumns = "[id], [payload], [tags]";
```

Comment:
looks like this var is not used in this method, can I remove it?

Reply:
No, embedding is used as the vector search at line 353 on the same method.

Reply:
sorry I mean the `queryColumns` var, it's not used, is it?

Reply:
My bad, you're identifying a bug in fact. I should use this var but it's in fact hard coded and the end... please let me fix it.